oauth2_provider/models.py

```diff
@@ -501,16 +502,27 @@ def revoke(self):
         """
         access_token_model = get_access_token_model()
         refresh_token_model = get_refresh_token_model()
-        with transaction.atomic():
+
+        access_token_database = router.db_for_write(access_token_model)
+        refresh_token_database = router.db_for_write(refresh_token_model)
+
+        # This is highly unlikely, but let's warn people just in case it does.
+        if access_token_database != refresh_token_database:
+            logger.warning(
+                "access token and refresh token are in separate databases but a transaction"
+                " is only used for the access token"
+            )
+
+        # Use the access_token_database instead of making the assumption it is in 'default'.
+        with transaction.atomic(using=access_token_database):
             token = refresh_token_model.objects.select_for_update().filter(pk=self.pk, revoked__isnull=True)
             if not token:
                 return
             self = list(token)[0]
 
-            try:
+            with suppress(access_token_model.DoesNotExist):
```

Comment:
I learned something new today!

Reply:
yeah, I learned about it like maybe a year ago? It was adding in Python 3.3 or 3.4 . It has become something I recommend in PR reviews. The explicit clarity of "I don't care" is really nice.